packages/react/time-series/src/utils.ts

```diff
@@ -213,3 +228,13 @@ export function getDefaultGranularity(timeRange?: TimeRangeInput) {
     [RelativeTimeRange.Last_90Days]: TimeSeriesGranularity.Day
   }[relative]
 }
+
+interface FormatLabelsOptions {
+  labels?: string[]
+  formatter?: (labels: string[]) => string[]
+}
+
+export function formatLabels(options: FormatLabelsOptions): string[] {
+  const { labels = [], formatter } = options
```

Comment:
Here we need to check if the formatter is a function for non-ts apps.

`typeof formatter === 'function'`

then:

`throw new Error(<MESSAGE>)`